op-e2e/system_test.go

```diff
@@ -840,3 +841,122 @@ func TestWithdrawals(t *testing.T) {
+	require.Nil(t, err, "sending gpo update tx")
+
+	receipt, err = waitForTransaction(tx.Hash(), l2Verif, 10*time.Duration(cfg.L1BlockTime)*time.Second)
+	require.Nil(t, err, "waiting for gpo decimals update tx")
+	require.Equal(t, receipt.Status, types.ReceiptStatusSuccessful, "transaction failed")
+
+	// Update scalar
+	tx, err = gpoContract.SetScalar(l2opts, big.NewInt(1_000_000))
+	require.Nil(t, err, "sending gpo update tx")
+
+	receipt, err = waitForTransaction(tx.Hash(), l2Verif, 10*time.Duration(cfg.L1BlockTime)*time.Second)
+	require.Nil(t, err, "waiting for gpo scalar update tx")
+	require.Equal(t, receipt.Status, types.ReceiptStatusSuccessful, "transaction failed")
+
+	overhead, err := gpoContract.Overhead(&bind.CallOpts{})
+	require.Nil(t, err, "reading gpo overhead")
+	decimals, err := gpoContract.Decimals(&bind.CallOpts{})
+	require.Nil(t, err, "reading gpo decimals")
+	scalar, err := gpoContract.Scalar(&bind.CallOpts{})
+	require.Nil(t, err, "reading gpo scalar")
+
+	require.Equal(t, overhead.Uint64(), uint64(2100), "wrong gpo overhead")
+	require.Equal(t, decimals.Uint64(), uint64(6), "wrong gpo decimals")
+	require.Equal(t, scalar.Uint64(), uint64(1_000_000), "wrong gpo scalar")
+
+	// Fee Recipient
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	feeStartBalance, err := l2Seq.BalanceAt(ctx, cfg.FeeRecipient, nil)
+	require.Nil(t, err)
+
+	// Simple transfer from signer to random account
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	startBalance, err := l2Verif.BalanceAt(ctx, fromAddr, nil)
+	require.Nil(t, err)
+
+	toAddr := common.Address{0xff, 0xff}
+	transferAmount := big.NewInt(1_000_000_000)
+	tx = types.MustSignNewTx(ethPrivKey, types.LatestSignerForChainID(cfg.L2ChainID), &types.DynamicFeeTx{
+		ChainID:   cfg.L2ChainID,
+		Nonce:     3, // Already have deposit
+		To:        &toAddr,
+		Value:     transferAmount,
+		GasTipCap: big.NewInt(0),
+		GasFeeCap: big.NewInt(200),
+		Gas:       21000,
+	})
+	err = l2Seq.SendTransaction(context.Background(), tx)
+	require.Nil(t, err, "Sending L2 tx to sequencer")
+
+	_, err = waitForTransaction(tx.Hash(), l2Seq, 3*time.Duration(cfg.L1BlockTime)*time.Second)
+	require.Nil(t, err, "Waiting for L2 tx on sequencer")
+
+	receipt, err = waitForTransaction(tx.Hash(), l2Verif, 3*time.Duration(cfg.L1BlockTime)*time.Second)
+	require.Nil(t, err, "Waiting for L2 tx on verifier")
+	require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status, "TX should have succeeded")
+
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	endBalance, err := l2Seq.BalanceAt(ctx, fromAddr, nil)
+	require.Nil(t, err)
+
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	feeEndBalance, err := l2Seq.BalanceAt(ctx, cfg.FeeRecipient, nil)
+	require.Nil(t, err)
+
+	feeDiff := new(big.Int).Sub(feeEndBalance, feeStartBalance)
+
+	// Calculate fee
```

Comment:
This is ensuring that the total balance in the system stays the same. You also need to look at the coinbase address. In addition it would be good to check that each fee amount is as expected.

Reply:
Added additional assertions for each fee amount. Also added coinbase fee check.